integrationtests/all-embedded-query-it/src/test/java/org/infinispan/all/embeddedquery/QueryDslConditionsTest.java

```diff
@@ -3093,4 +3093,30 @@ public void testTwoPhaseGroupingAndAggregationOnSameField() {
       assertEquals(1L, list.get(2)[0]);
       assertNull(list.get(2)[1]);
    }
+
+   @Test
+   public void testLuceneWildcardsAreEscaped() {
+      QueryFactory qf = getQueryFactory();
+      Query q = qf.from(getModelFactory().getUserImplClass())
+            .having("name").like("J%n").toBuilder()
+            .build();
+
+      assertEquals(1, q.list().size());
+
+      q = qf.from(getModelFactory().getUserImplClass())
```

Comment:
The query is executed again. But let me use another variable name to make it clearer.